### PR TITLE
[LEAK-INFORMED] match the rest of pokedex_screen

### PR DESCRIPTION
**Pull request by Kurausukun**
Many thanks to seviper and luckytyphlosion on discord, who helped match the functions with changes.

### CODE REVIEW

File: include/pokedex_screen.h [hunk 1]
```diff
@@ -11,6 +11,11 @@
 #define DEX_CATEGORY_URBAN         7
 #define DEX_CATEGORY_RARE          8
 
+#define DEX_CATEGORY_SIZE          (sizeof(gDexCategories)/sizeof(struct {                                      \
+                                                                      const struct PokedexCategoryPage * page;  \
+                                                                      u8 count;                                 \
+                                                                  }))
+
```
Comment (GriffinRichards) [inline include/pokedex_screen.h:18]: No need for this. `NELEMS(gDexCategories)`

File: src/pokedex_screen.c [hunk 2]
```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
...
+        if (inches % 10 >= 5)
+            inches += 10;
+        feet = inches / 120;
+        inches = (inches - (feet * 120)) / 10;
+        if (feet / 10 == 0)
+        {
+            buffer[i++] = 0;
+            buffer[i++] = feet + CHAR_0;
+        }
+
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:2672]: Remove whitespace above `else` (below as well)

File: src/pokedex_screen.c [hunk 3]
```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
...
+            i++;
+        }
+
+        lbs %= 1000;
+        buffer[i++] = (lbs / 100) + CHAR_0;
+        lbs %= 100;
+        buffer[i++] = CHAR_PERIOD;
+        buffer[i++] = (lbs / 10) + CHAR_0;
+    }
+
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:2769]: More whitespace above `else`

File: src/pokedex_screen.c [hunk 4]
```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
...
+    sub_8105D64(gUnknown_203ACF0->field_4A[1], gUnknown_203ACF0->field_5A, 88, 40);
+
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[1]);
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[1], 2);
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[2], 0);
+    sub_8105CB0(gUnknown_203ACF0->field_4A[2], gUnknown_203ACF0->field_5A, 0, 8);
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[2]);
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[2], 2);
+    FillWindowPixelBuffer(1, 255);
+    if (!a0)
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:2892]: If `a0` is `bool8` update the argument too (otherwise do `== 0`)

File: src/pokedex_screen.c [hunk 5]
```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
...
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[10], 2);
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[8], 0);
+    sub_8104A34(gUnknown_203ACF0->field_4A[8], 0, species, 0, 0);
+    sub_81047C8(gUnknown_203ACF0->field_4A[8], 2, gSpeciesNames[species], 3, 12, 0);
+    PutWindowTilemap(gUnknown_203ACF0->field_4A[8]);
+    CopyWindowToVram(gUnknown_203ACF0->field_4A[8], 2);
+    FillWindowPixelBuffer(gUnknown_203ACF0->field_4A[12], 0);
+    ListMenuLoadStdPalAt(176, 1);
+
+    if (v3)
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:3011]: Same here, `v3` as a bool

File: src/pokedex_screen.c [hunk 6]
```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
...
+}
+
+u8 sub_8106B60(u16 species)
+{
+    sub_8104AB0(species, 2, 1);
+    sub_8104AB0(species, 3, 1);
+
+    if (!IsNationalPokedexEnabled())
+        if (SpeciesToNationalPokedexNum(species) > KANTO_DEX_COUNT)
+            return CreateTask(sub_8106BD8, 0);
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:3232]: Does `if (!IsNationalPokedexEnabled() && SpeciesToNationalPokedexNum(species) > KANTO_DEX_COUNT)` match?

File: src/pokedex_screen.c [hunk 7]
```diff
@@ -2541,3 +2546,827 @@ static bool8 sub_81052D0(u8 a0)
...
+    count = gDexCategories[categoryNum].page[pageNum].count;
+
+    for (i = 0; i < 4; i++)
+    {
+        if (i < count)
+        {
+            species = gDexCategories[categoryNum].page[pageNum].species[i];
+            if (sub_8106810(species) == TRUE)
+                if (sub_8104AB0(species, 0, 1))
+                    return 1;
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:3105]: `if (sub_8106810(species) == TRUE && sub_8104AB0(species, 0, 1))`? Same in other uses of `sub_8106810`

**Review by GriffinRichards — CHANGES_REQUESTED** [review]

File: src/pokedex_screen.c (hunk 5 above)
Comment (Kurausukun) [reply src/pokedex_screen.c:3011]: Changed to bool8, but `== TRUE` doesn't match sadly

File: src/pokedex_screen.c (hunk 6 above)
Comment (Kurausukun) [reply src/pokedex_screen.c:3232]: It does, changed

File: src/pokedex_screen.c (hunk 5 above)
Comment (GriffinRichards) [reply src/pokedex_screen.c:3011]: Yeah as is is fine, `== TRUE` is different than `!= FALSE`

File: src/pokedex_screen.c (hunk 7 above)
Comment (Kurausukun) [reply src/pokedex_screen.c:3105]: Done

File: include/pokedex_screen.h (hunk 1 above)
Comment (Kurausukun) [reply include/pokedex_screen.h:18]: Oops, forgot that existed

Comment (GriffinRichards) [reply include/pokedex_screen.h]: Sorry should've been clearer, the usage of `DEX_CATEGORY_SIZE` itself can just be replaced with `NELEMS(gDexCategories)`.

File: src/pokedex_screen.c [hunk 8]
```diff
@@ -2541,3 +2546,817 @@ static bool8 sub_81052D0(u8 a0)
...
+        PlayCry2(gUnknown_203ACF0->field_5A, 0, 125, 10);
+}
+
+u8 sub_8106B60(u16 species)
+{
+    sub_8104AB0(species, 2, 1);
+    sub_8104AB0(species, 3, 1);
+
+    if (!IsNationalPokedexEnabled() && SpeciesToNationalPokedexNum(species) > KANTO_DEX_COUNT)
+            return CreateTask(sub_8106BD8, 0);
```
Comment (GriffinRichards) [inline src/pokedex_screen.c:3222]: Fix spacing

**Review by GriffinRichards — APPROVED** [review]

**Pull request merged**